src/VirtoCommerce.CustomerModule.Data/Search/Indexing/MemberDocumentBuilder.cs

```diff
@@ -27,22 +27,19 @@ public MemberDocumentBuilder(IMemberService memberService, IDynamicPropertySearc
         public virtual async Task<IList<IndexDocument>> GetDocumentsAsync(IList<string> documentIds)
         {
             var members = await GetMembers(documentIds);
-            var result = new List<IndexDocument>();
-
-            foreach (var member in members)
-            {
-                result.Add(await CreateDocument(member));
-            }
 
+            IList<IndexDocument> result = members.Select(CreateDocument).ToArray();
             return result;
         }
 
-        protected virtual Task<Member[]> GetMembers(IList<string> documentIds)
+        protected virtual async Task<Member[]> GetMembers(IList<string> documentIds)
         {
-            return _memberService.GetByIdsAsync(documentIds.ToArray());
+            var result = await _memberService.GetByIdsAsync(documentIds.ToArray());
+            await LoadDependencies(result);
+            return result;
         }
 
-        protected async virtual Task<IndexDocument> CreateDocument(Member member)
+        protected virtual IndexDocument CreateDocument(Member member)
```

Comment:
We already changed our module to override the async version.
You should not revert to sync again.

Reply:
@artem-dudarev What about others, who didn't upgraded already?

Reply:
we can add 'Async' method